client/src/main/java/org/eclipse/hono/client/CommandClient.java

```diff
@@ -24,11 +24,10 @@
      *         The future will succeed if a response with status 201 has been received from the device.
      *         <p>
      *         Otherwise, the future will fail with a {@link ServiceInvocationException} containing
-     *         the (error) status code.
+     *         the (error) status code. Status codes are defined at <a href="https://www.eclipse.org/hono/api/command-and-control-api"/>
```

Comment:
I still do not believe that the future will succeed if the device responded with a 201 status code.

Reply:
It should be 200.

Reply:
Shouldn't this depend on the *nature* of the command, e.g. if the response contains payload (200) or not (204) or whether the command was used to create a resource (201) or not?

Reply:
What about (from API definition)?

| *2xx* | Indicating success; the codes are dependent on the adapter and device, which handles the command. |
| *4xx* | Indicating failure; the codes are dependent on the adapter and device, which handles the command. |
| *503* | The system responsible of handling the command is not available. |

Reply:
I would replace the last description with
> The command cannot be sent to the target device.

Reply:
Okay